subcommands/devices/show.go

```diff
@@ -64,6 +65,17 @@ func doShow(cmd *cobra.Command, args []string) {
 	if len(device.CurrentUpdate) > 0 {
 		fmt.Printf("Update Id:\t%s\n", device.CurrentUpdate)
 	}
+	if device.AppsState != nil && device.AppsState.Apps != nil {
+		stateColor := map[string]color.Attribute{
+			"healthy":   color.FgGreen,
+			"unhealthy": color.FgRed,
+		}
+		var appStates []string
+		for a, s := range device.AppsState.Apps {
+			appStates = append(appStates, color.New(stateColor[s.State]).Sprintf("%s:%s", a, s.State))
+		}
+		fmt.Printf("App states:\t%s\n", strings.Join(appStates, ", "))
```

Comment:
I would prefer this to be a vertically aligned list instead of comma-separated one, like I pledged earlier.

This is not a critical comment, just MHO.

Reply:
It looks a regular terminal on a regular screen (14, 15 inches) can accommodate around 7 apps. Of course, it depends on length of app names, since customers usually have just a few apps it looks neater to me to render them horizontally.

Reply:
ATM we have 17 production devices with more than 7 apps (50 with more than 5):
- one belongs to Schneider;
- around 10 belongs to different Arduino factories (I have no clue who owns them) - these are extremely long;
- other are most probably yours.

Also, @doanac said yesterday that we have some customer who is going to install more than a dozen of docker apps to their devices.
So, devices with the relatively big number of apps is the reality.

```
    "repo 84d6f7e1-837c-4b06-b43a-bc110f368aff (last seen 2020-06-15T16:04:19Z) -> shellhttpd, app-01, helloworld, app-02, app-03, app-04, app-05, app-06",
    "repo 84d6f7e1-837c-4b06-b43a-bc110f368aff (last seen 2020-06-15T16:22:44Z) -> shellhttpd, app-01, helloworld, app-02, app-03, app-04, app-05, app-06",
    "repo 84d6f7e1-837c-4b06-b43a-bc110f368aff (last seen 2020-06-15T16:02:57Z) -> shellhttpd, app-01, helloworld, app-02, app-03, app-04, app-05, app-06",
    "repo 84d6f7e1-837c-4b06-b43a-bc110f368aff (last seen 2020-06-15T16:16:56Z) -> shellhttpd, app-01, helloworld, app-02, app-03, app-04, app-05, app-06",
    "repo 84d6f7e1-837c-4b06-b43a-bc110f368aff (last seen 2020-06-15T08:22:51Z) -> shellhttpd, app-01, helloworld, app-02, app-03, app-04, app-05, app-06",
    "repo 84d6f7e1-837c-4b06-b43a-bc110f368aff (last seen 2020-06-15T16:02:00Z) -> shellhttpd, app-01, helloworld, app-02, app-03, app-04, app-05, app-06",
    "repo d7dc8032-f026-491c-bdf9-55aa45991d7e (last seen 2021-10-07T18:29:30Z) -> alpine-dev, cli, eclipse-mosquitto, fiotest, led-control, ota-manager, redis, ssh-container, wiser-home",
    "repo cd08a18f-252a-47a8-bd7d-168ecf0f0ebc (last seen 2022-05-23T03:10:45.438Z) -> arduino-ide, arduino-ootb, blob-opera, cloudshell, fiotest, go-rpc-proxy, go-rpc-serial, go-wifi-captive-portal, iot-trinity, python-hello-world, python-rpc-example, rpi-flask-webserver, x-kiosk",
    "repo d8841a79-c00a-449f-8d2e-486d3cb1e03f (last seen 2022-06-01T09:10:50.218Z) -> arduino-ide, arduino-ootb, blob-opera, cloudshell, fiotest, go-rpc-proxy, go-rpc-serial, go-wifi-captive-portal, iot-trinity, python-hello-world, python-rpc-example, rpi-flask-webserver, x-kiosk",
    "repo f8abb388-17cd-4e6d-9df6-d22068276f29 (last seen 2022-03-24T12:53:25.124Z) -> arduino-ide, arduino-ootb, blob-opera, cloudshell, fiotest, go-rpc-proxy, go-rpc-serial, go-wifi-captive-portal, iot-trinity, python-hello-world, python-rpc-example, rpi-flask-webserver, x-kiosk",
    "repo 312d6a5d-b8e6-46bd-b23e-1f6b0f1401ac (last seen 2022-07-11T05:30:18.568Z) -> arduino-ide, arduino-ootb, blob-opera, cloudshell, fiotest, go-rpc-proxy, go-rpc-serial, go-wifi-captive-portal, iot-trinity, python-hello-world, python-rpc-example, rpi-flask-webserver, x-kiosk",
    "repo 4bc893df-272c-4079-aded-fd1136d19cdb (last seen 2022-07-31T16:51:15.442Z) -> arduino-ide, arduino-ootb, blob-opera, cloudshell, fiotest, go-rpc-proxy, go-rpc-serial, go-wifi-captive-portal, iot-trinity, python-hello-world, python-rpc-example, rpi-flask-webserver, x-kiosk",
    "repo 57814f4a-98f7-4631-97cb-a90feab19411 (last seen 2022-06-08T13:44:53.616Z) -> arduino-ide, arduino-ootb, blob-opera, cloudshell, fiotest, go-rpc-proxy, go-rpc-serial, go-wifi-captive-portal, iot-trinity, python-hello-world, python-rpc-example, rpi-flask-webserver, x-kiosk",
    "repo 7c73bb8a-8a2a-415f-9880-5500628f6f0b (last seen 2022-05-29T03:53:41.893Z) -> arduino-ide, arduino-ootb, blob-opera, cloudshell, fiotest, go-rpc-proxy, go-rpc-serial, go-wifi-captive-portal, iot-trinity, python-hello-world, python-rpc-example, rpi-flask-webserver, x-kiosk",
    "repo b2565d16-fed0-4c0b-bb81-3e2f47e407d5 (last seen 2022-07-20T19:57:14.633Z) -> arduino-ide, arduino-ootb, blob-opera, cloudshell, fiotest, go-rpc-proxy, go-rpc-serial, go-wifi-captive-portal, iot-trinity, python-hello-world, python-rpc-example, rpi-flask-webserver, x-kiosk",
    "repo b7c92b96-1d6e-4351-9810-3ac7a50d4fac (last seen 2022-04-05T14:34:32.224Z) -> arduino-ide, arduino-ootb, blob-opera, cloudshell, fiotest, go-rpc-proxy, go-rpc-serial, go-wifi-captive-portal, iot-trinity, python-hello-world, python-rpc-example, rpi-flask-webserver, x-kiosk",
    "repo b7c92b96-1d6e-4351-9810-3ac7a50d4fac (last seen 2022-03-31T14:35:24.027Z) -> arduino-ide, arduino-ootb, blob-opera, cloudshell, fiotest, go-rpc-proxy, go-rpc-serial, go-wifi-captive-portal, iot-trinity, python-hello-world, python-rpc-example, rpi-flask-webserver, x-kiosk"
```

Try to run this query on our prod CouchBase node:
```
select raw "repo " || repo_id || " (last seen " || millis_to_str(last_seen * 1000, "YYYY-MM-DD") || ") -> " || CONCAT2(", ", docker_apps) from `device-meta` where doctype='device' and array_length(docker_apps) > 5;
```
It shows that even some devices with just 6 apps have some quite long app names.

Reply:
Only active within the last 3 weeks with more than 5 apps - 20 devices (2 with more than 10 apps):
```
cbq> select raw "repo " || repo_id || " (last seen " || millis_to_str(last_seen * 1000) || ") -> " || CONCAT2(", ", docker_apps) from `device-meta` where doctype='device' and array_length(docker_apps) > 5 and last_seen > 1658000000 order by array_length(docker_apps);

    "repo d7dc8032-f026-491c-bdf9-55aa45991d7e (last seen 2022-08-10T14:40:56.564Z) -> alpine-dev, inno2grid, mqtt, nginx, redis, wiser-home",
    "repo 84d6f7e1-837c-4b06-b43a-bc110f368aff (last seen 2022-08-10T08:28:03.385Z) -> app-03, app-04, app-05, app-06, one-shot, one-shot-failing",
    "repo caa5cc5c-787b-4b52-b53a-ce63da7e5a75 (last seen 2022-07-26T10:55:31.191Z) -> apps, datalog, eip, event_manager, invertermanager, modbus",
    "repo caa5cc5c-787b-4b52-b53a-ce63da7e5a75 (last seen 2022-08-09T16:39:10.04Z) -> apps, eip, event_manager, gw_bms_interface, invertermanager, modbus",
    "repo caa5cc5c-787b-4b52-b53a-ce63da7e5a75 (last seen 2022-07-21T23:24:24.744Z) -> apps, eip, event_manager, gw_bms_interface, invertermanager, modbus",
    "repo caa5cc5c-787b-4b52-b53a-ce63da7e5a75 (last seen 2022-07-26T10:29:04.762Z) -> apps, datalog, eip, event_manager, invertermanager, modbus",
    "repo caa5cc5c-787b-4b52-b53a-ce63da7e5a75 (last seen 2022-08-10T14:38:59.322Z) -> apps, eip, event_manager, gw_bms_interface, invertermanager, modbus",
    "repo caa5cc5c-787b-4b52-b53a-ce63da7e5a75 (last seen 2022-08-03T16:57:59.817Z) -> apps, eip, event_manager, gw_bms_interface, invertermanager, modbus",
    "repo caa5cc5c-787b-4b52-b53a-ce63da7e5a75 (last seen 2022-08-10T14:38:16.581Z) -> apps, eip, event_manager, gw_bms_interface, invertermanager, modbus",
    "repo caa5cc5c-787b-4b52-b53a-ce63da7e5a75 (last seen 2022-08-08T22:50:07.804Z) -> apps, eip, event_manager, gw_bms_interface, invertermanager, modbus",
    "repo caa5cc5c-787b-4b52-b53a-ce63da7e5a75 (last seen 2022-08-09T21:24:36.339Z) -> apps, eip, event_manager, gw_bms_interface, invertermanager, modbus",
    "repo caa5cc5c-787b-4b52-b53a-ce63da7e5a75 (last seen 2022-08-04T12:01:29.471Z) -> apps, eip, event_manager, gw_bms_interface, invertermanager, modbus",
    "repo caa5cc5c-787b-4b52-b53a-ce63da7e5a75 (last seen 2022-07-26T09:40:34.674Z) -> apps, datalog, eip, event_manager, invertermanager, modbus",
    "repo caa5cc5c-787b-4b52-b53a-ce63da7e5a75 (last seen 2022-08-04T17:58:08.415Z) -> apps, eip, event_manager, gw_bms_interface, invertermanager, modbus",
    "repo caa5cc5c-787b-4b52-b53a-ce63da7e5a75 (last seen 2022-07-28T10:38:13.519Z) -> apps, datalog, eip, event_manager, invertermanager, modbus",
    "repo caa5cc5c-787b-4b52-b53a-ce63da7e5a75 (last seen 2022-08-04T15:45:21.178Z) -> apps, eip, event_manager, gw_bms_interface, invertermanager, modbus",
    "repo caa5cc5c-787b-4b52-b53a-ce63da7e5a75 (last seen 2022-07-20T04:02:37.466Z) -> apps, datalog, eip, event_manager, invertermanager, modbus",
    "repo caa5cc5c-787b-4b52-b53a-ce63da7e5a75 (last seen 2022-07-19T18:24:08.488Z) -> apps, eip, event_manager, gw_bms_interface, invertermanager, modbus",
    "repo b2565d16-fed0-4c0b-bb81-3e2f47e407d5 (last seen 2022-07-20T19:57:14.633Z) -> arduino-ide, arduino-ootb, blob-opera, cloudshell, fiotest, go-rpc-proxy, go-rpc-serial, go-wifi-captive-portal, iot-trinity, python-hello-world, python-rpc-example, rpi-flask-webserver, x-kiosk",
    "repo 4bc893df-272c-4079-aded-fd1136d19cdb (last seen 2022-07-31T16:51:15.442Z) -> arduino-ide, arduino-ootb, blob-opera, cloudshell, fiotest, go-rpc-proxy, go-rpc-serial, go-wifi-captive-portal, iot-trinity, python-hello-world, python-rpc-example, rpi-flask-webserver, x-kiosk"
```

Reply:
> yesterday that we have some customer who is going to install more than a dozen of docker apps to their devices.

iiuc, he meant images not apps.

Reply:
i don't really like this style. But then i see we do the Docker Apps this way already. So -i'd just do this, or go higher level and just this up into a "total" of: `App states: healthy|unhealthy`

Reply:
In addition to that `devices list` print comma-separated list of apps as a column element which looks on terminal even worse.

```
msul-local-103                                       unknown                                                                           OFFLINE                              shellhttpd,app-01,helloworld,app-02,app-03,app-04,app-05         false       false
```

Reply:
I think I'll just output two rows, `Healthy Apps: <>` and `Unhealthy Apps: <>`, so it will be in line with `Docker Apps:` output.

Reply:
Or, we actually can tweak an existing field `Docker Apps` so it list Apps along with its state. However, the Apps list not necessarily will be the same.

Reply:
> I think I'll just output two rows, Healthy Apps: <> and Unhealthy Apps: <>, so it will be in line with Docker Apps: output.

This sounds like a good idea to me; at the least it saves a lot of space used by "healthy|unhealthy" clobbering.
Furthermore, as we already have the _**Docker Apps**_ field listing _all_ apps, it looks like printing just _**Unhealthy Apps**_ is enough.

Reply:
I thought about it too, the problem is that`Docker Apps` set is not always equal to sum of healthy and unhealthy apps. The first set is what we installed the second one is what docker engine reports.

Reply:
yep - i wanted to do this, but i think having all three lists is best and might really help if something crazy is happening on a device.

Reply:
makes sense 👍